src/persistence/subject-builder/OneToOneInverseSideSubjectBuilder.ts

```diff
@@ -166,6 +166,7 @@ export class OneToOneInverseSideSubjectBuilder {
                     metadata: relation.inverseEntityMetadata,
                     canBeUpdated: true,
                     identifier: relationIdMap,
+                    entity: relatedEntity,
```

Comment:
This would also be helpful for many-to-many relationships: https://github.com/typeorm/typeorm/blob/58fc08840a4a64ca1935391f4709a784c3f0b373/src/persistence/subject-builder/ManyToManySubjectBuilder.ts#L172-L175